Storage/SQL/SQLiteBookmarks.swift

```diff
@@ -166,30 +166,36 @@ public class SQLiteBookmarks: BookmarksModelFactory {
     }
 
     public func clearBookmarks() -> Success {
-        return self.db.run("DELETE FROM \(TableBookmarks) WHERE parent IS NOT \(BookmarkRoots.RootID)")
-    }
-
-    private func runSQL(sql: String, args: Args?, success: (Bool) -> Void, failure: (Any) -> Void) {
         var err: NSError?
-        self.db.withWritableConnection(&err) { (connection: SQLiteDBConnection, inout err: NSError?) -> Int in
-            if let err = connection.executeChange(sql, withArgs: args) {
-                failure(err)
-                return 0
+        return self.db.withWritableConnection(&err) { (connection: SQLiteDBConnection, inout err: NSError?) -> Success in
+            if let err = connection.executeChange("DELETE FROM \(TableBookmarks) WHERE parent IS NOT ?", withArgs: [BookmarkRoots.RootID]) {
+                return deferResult(DatabaseError(err: err))
+            }
+            if let err = self.favicons.removeUnused(connection) {
```

Comment:
Need to figure out a neater way to run multiple DB ops with the same connection. `self.db.run()` is too nice to lose :/